### PR TITLE
feat: 🎸 add debounce to SQForm onSubmit and buttons onClick

### PR DESCRIPTION
**Pull request by codyarose**
Add [debounced callback](https://www.npmjs.com/package/use-debounce) to the `SQForm` onSubmit and `SQFormButton`/`SQFormIconButton` onClick. For the buttons the debounce is handled within `useFormButton`.

The timer is set to 500ms, as discussed in the issue, and the config is set to allow leading calls, but not trailing. Meaning the event will be fired immediately upon initiation, then can't be fired again until the timer has completed.

Loom: https://www.loom.com/share/e2522c52a8b9422b9c969359a06359c2

✅ Closes: #123

### CODE REVIEW

File: src/components/SQForm/SQFormButton.js [hunk 1]
```diff
@@ -28,7 +30,9 @@ function SQFormButton({
       title={title}
       type={type}
       isDisabled={isSQFormButtonDisabled}
-      onClick={type === 'reset' ? handleReset : undefined}
+      onClick={
+        type === 'reset' ? handleReset : onClick ? handleClick : undefined
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormButton.js:34]: I'm not the biggest fan of nested ternaries. This one isn't terrible. Would be possible do a `getClickHandler` to make determining the click handler for this button more readable? If others don't think this is necessary that's fine.

Comment (codyarose) [reply src/components/SQForm/SQFormButton.js:34]: Yeah, that's fair. I'll add a fix

**Review by 20BBrown14 — APPROVED** [review]

**Comment by SeanGroff** [comment]
@codyarose Can you provide a Loom video to show the expected functionality within a few stories still functions?
Bonus points if you can show the debounce working.

**Comment by codyarose** [comment]
> @codyarose Can you provide a Loom video to show the expected functionality within a few stories still functions?
> Bonus points if you can show the debounce working.

👍 https://www.loom.com/share/e2522c52a8b9422b9c969359a06359c2

**Review by SeanGroff — APPROVED** [review]
Great work @codyarose 
Great idea @aleerobinson

**Pull request merged**